internal/worker/json.go

```diff
@@ -256,6 +256,29 @@ type ContainerResolveJobResult struct {
 	JobResult
 }
 
+type OSTreeResolveSpec struct {
+	URL    string `json:"url"`
+	Ref    string `json:"ref"`
+	Parent string `json:"parent"`
+}
+
+// TODO better name
+type OSTreeResolvedSpec struct {
+	URL      string `json:"url"`
+	Ref      string `json:"ref"`
+	Checksum string `json:"checksum"`
+}
+
+type OSTreeResolveJob struct {
+	Specs []OSTreeResolveSpec `json:"ostree_resolve_specs"`
```

Comment:
@achilleas-k I noticed in `internal/distro` there's some indication of supporting multiple commits. But the `ImageOptions` don't account for this at all. 

Should I bother with a slice here to make any switch to multiple commits easier in a worker/job api?

Reply:
That's probably a good idea.  I also saw that @teg added multiple ostree commit support to the Pipeline object, but I'm not sure what the use case for this is exactly.  If it's for pipelines that can **build** multiple commits, that's one thing.  We would only need to resolve multiple ostree specs if we have multiple parents and I'm not sure we have a use case for that.

All that said, having a slice in the job args and only ever using one element can't hurt, right?


https://github.com/osbuild/osbuild-composer/blob/a23b1c4b3e414d80600d19c5fc81659b4f1a24c1/internal/manifest/pipeline.go#L29